pkgs/development/python-modules/pyelftools/default.nix

```
@@ -1,31 +1,30 @@
 { stdenv
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
 , python
 }:
 
 buildPythonPackage rec {
   pname = "pyelftools";
-  version = "0.26";
+  version = "unstable-2020-09-23";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "86ac6cee19f6c945e8dedf78c6ee74f1112bd14da5a658d8c9d4103aed5756a2";
+  src = fetchFromGitHub {
+    owner = "eliben";
+    repo = pname;
+    rev = "ab84e68837113b2d700ad379d94c1dd4a73125ea";
+    sha256 = "sha256-O7l1kj0k8bOSOtZJVzS674oVnM+X3oP00Ybs0qjb64Q=";
   };
 
   checkPhase = ''
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" test/external_tools/readelf
     ${python.interpreter} test/all_tests.py
```

looks like other packages are overriding this and pulling from pypi, in which this doesn exist:
```suggestion
    if [ -x test/external_tools/readelf ]; then
       patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" test/external_tools/readelf
       ${python.interpreter} test/all_tests.py
    fi
```

---

I would also add a `pythonImportsCheck`

---

```
https://github.com/NixOS/nixpkgs/pull/99202
2 packages failed to build:
esphome platformio

7 packages built:
pwndbg pwntools python27Packages.pwntools python27Packages.pyelftools python37Packages.pwntools python37Packages.pyelftools python38Packages.pyelftools
```